File: java/src/sliding_window/LC_1343_NumberOfSubArraysAverageThreshold.java
package sliding_window;

// https://leetcode.com/problems/number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold
public class LC_1343_NumberOfSubArraysAverageThreshold {
  public int numOfSubarrays(int[] arr, int k, int threshold) {
    int subArraysCount = 0, sum = 0, minSum = k * threshold;
    for (int i = 0; i < k; i++) {
      sum += arr[i];
    }
    if (sum >= minSum) {
      subArraysCount++;
    }

    for (int j = k; j < arr.length; j++) {
      sum += arr[j] - arr[j - k];
      if (sum > minSum) {
        subArraysCount++;
      }
    }
    return subArraysCount;
  }
}

File: python/sliding_window/LC_1343_NumberOfSubArraysAverageThreshold.py
# https://leetcode.com/problems/number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/
from typing import List

class Solution:
    def numOfSubarrays(self, arr: List[int], k: int, threshold: int) -> int:
        total = 0
        for num in arr[:k]:
            total += num
        arr_count = 0
        if total // k >= threshold:
            arr_count += 1
        
        for i in range(k, len(arr)):
            total -= arr[i-k]
            total += arr[i]
            if total // k >= threshold:
                arr_count += 1
        return arr_count
    
    def numOfSubarrays_2(self, arr: List[int], k: int, threshold: int) -> int:
        total = 0
        min_sum = k * threshold
        for num in arr[:k]:
            total += num
        arr_count = 0
        if total >= min_sum:
            arr_count += 1
        
        for i in range(k, len(arr)):
            total += arr[i] - arr[i-k]
            if total >= min_sum:
                arr_count += 1
        return arr_count

